static-website-azure-python/__main__.py

```diff
@@ -0,0 +1,73 @@
+import pulumi
+import pulumi_azure_native as azure_native
+import pulumi_synced_folder as synced_folder
+import urllib
```

Comment:
Are you specifically trying to use only built-in libraries? I generally only find `urllib` versus `requests` or `urllib3` when people are trying to restrict their dependency/security surface area.

Reply:
Also, if you're only using one method or class, it's generally better practice to import just that need to avoid polluting your namespace, similar to the following:
```
from urllib.parse import urlparse
```

Finally, if you're going to import built-ins, generally PEP8 style has built-ins first (standard library modules), then third-party, then local (https://peps.python.org/pep-0008/#imports).

Reply:
>Are you specifically trying to use only built-in libraries?

Ideally (and I guess in general) yes; assuming this works and isn't somehow problematic, I figure folks can opt into a different library if they want to.

Reply:
> PEP8 style has built-ins first (standard library modules), then third-party, then local

Ok thanks. So like this?

```python
from urllib.parse import urlparse
import pulumi
import pulumi_azure_native as azure_native
import pulumi_synced_folder as synced_folder
```

Reply:
> > Are you specifically trying to use only built-in libraries?
> 
> Ideally (and I guess in general) yes; assuming this works and isn't somehow problematic, I figure folks can opt into a different library if they want to.

Makes sense; it *is* a common practice if you're trying to reduce surface area. I probably would leave a comment in the line, then, just so newer Python folks aren't confused (most tutorials/self-guided learning systems use requests or urllib3; they very rarely use the built-in. If you want to explore why, check out https://realpython.com/urllib-request/#the-request-package-ecosystem, which has a very good writeup). 

If you add a comment, two spaces after the end of the import will make every linter happy :)

```python
import package  # Here's a comment
```

Reply:
> > PEP8 style has built-ins first (standard library modules), then third-party, then local
> 
> Ok thanks. So like this?
> 
> ```python
> from urllib.parse import urlparse
> import pulumi
> import pulumi_azure_native as azure_native
> import pulumi_synced_folder as synced_folder
> ```

Yep, that technically meets PEP8. A lot of folks follow a slightly modified version that puts the `from .. import ..` calls at the bottom in the same order. Let me add a couple packages to show you what I mean:

```python
import os
import sys

import pulumi
import pulumi_azure_native as azure_native
import pulumi_synced_folder as synced_folder

from urllib.parse import urlparse

from requests.exceptions import HTTPError
```

**However**, that gets into arguments as to which style to follow that are more pedantic than helpful. As long as you follow PEP8 and the code is readable, you're good.